packages/core/src/editor/useInternalEditor.ts

```diff
@@ -26,22 +26,33 @@ export type useInternalEditorReturnType<C = null> = useCollectorReturnType<
 > & {
   inContext: boolean;
   store: EditorStore;
-  connectors: ChainableConnectors<
-    CoreEventHandlers['connectors'],
-    React.ReactElement
-  >;
+  connectors: EventHandlerConnectors<CoreEventHandlers, React.ReactElement>;
 };
 
 export function useInternalEditor<C>(
   collector?: EditorCollector<C>
 ): useInternalEditorReturnType<C> {
-  const handlers = useEventHandler();
+  const handler = useEventHandler();
   const store = useContext(EditorContext);
   const collected = useCollector(store, collector);
 
+  const connectorInstance = useMemo(() => handler.createConnectorInstance(), [
+    handler,
+  ]);
+
+  useEffect(() => {
+    return () => {
+      if (!connectorInstance) {
+        return;
+      }
+
+      connectorInstance.cleanup();
+    };
+  }, [connectorInstance]);
+
```

Comment:
Cleanup any connectors used when the component unmounts. Previously, we only removed the connectors when the editor is enabled/disabled.